apps/framework-cli/src/framework/core/infrastructure/function_process.rs

```diff
@@ -0,0 +1,145 @@
+        FunctionProcess {
+            name: function.name.clone(),
+
+            // This probably should be a reference to the topic ingested from the
+            // infra map insteaad of using a convention for the topic name.
+            // Leaving it as is for compatibility with the current code.
+            source_topic: get_latest_topic(topics, &function.source_data_model.name)
+                .unwrap_or_else(|| function.source_data_model.name.clone()),
+            source_columns: function.source_data_model.columns.clone(),
+
+            // This probably should be a reference to the topic ingested from the
+            // infra map insteaad of using a convention for the topic name.
+            // Leaving it as is for compatibility with the current code.
+            target_topic: get_latest_topic(topics, &function.target_data_model.name)
+                .unwrap_or_else(|| function.target_data_model.name.clone()),
+
+            target_columns: function.target_data_model.columns.clone(),
+            target_topic_config: HashMap::from([
+                ("max.message.bytes".to_string(), (1024 * 1024).to_string()),
+                ("message.max.bytes".to_string(), (1024 * 1024).to_string()),
+            ]),
+
+            executable: function.executable.clone(),
+
+            version: function.version.clone(),
+            source_primitive: PrimitiveSignature {
+                name: function.name.clone(),
+                primitive_type: PrimitiveTypes::Function,
+            },
+        }
+    }
+
+    pub fn from_migration_functon(
+        function: &Flow,
+        source_topic: &Topic,
+        target_topic: &Topic,
+    ) -> Self {
+        FunctionProcess {
+            name: function.name.clone(),
+
+            source_topic: source_topic.name.clone(),
+            source_columns: function.source_data_model.columns.clone(),
+
+            target_topic: target_topic.name.clone(),
+
+            target_columns: function.target_data_model.columns.clone(),
+            target_topic_config: HashMap::from([
+                ("max.message.bytes".to_string(), (1024 * 1024).to_string()),
+                ("message.max.bytes".to_string(), (1024 * 1024).to_string()),
+            ]),
+
+            executable: function.executable.clone(),
+
+            version: function.version.clone(),
+            source_primitive: PrimitiveSignature {
+                name: function.name.clone(),
+                primitive_type: PrimitiveTypes::Function,
+            },
+        }
+    }
+
+    pub fn is_ts_function_process(&self) -> bool {
+        self.executable.extension().unwrap().to_str().unwrap() == TYPESCRIPT_FILE_EXTENSION
+    }
+
+    pub fn is_py_function_process(&self) -> bool {
+        self.executable.extension().unwrap().to_str().unwrap() == PYTHON_FILE_EXTENSION
+    }
+
+    pub fn id(&self) -> String {
+        format!(
+            "{}_{}_{}_{}",
+            self.name, self.source_topic, self.target_topic, self.version
```

Comment:
Is this the same as Flow.id? Wondering if it's safer to have an id field and we save the value when creating a FunctionProcess because we have the Flow object at that time so both ids are in sync.

Reply:
As of now it is but it is a good callout - it probably should be different since they are different objects.